src/vivarium_gates_nutrition_optimization_child/components/fertility.py

```diff
@@ -69,7 +73,18 @@ def on_time_step(self, event: Event) -> None:
         )
         # TODO: remove resetting index when using actual child data
         born_previous_step = birth_records[born_previous_step_mask].reset_index().copy()
+        # everyone is currently born on the first time step so this is always empty after the first time step
+        if born_previous_step.empty:
```

Comment:
I still think this should be refactored to be done on initialization but I guess as long as this works. Just seems pretty hacky

Reply:
Is this even needed? Seems like these would all be no-ops on an empty `born_previous_step` maybe?

Reply:
@stevebachmeier I agree that it's inefficient/hacky but it would unfortunately require a significant refactor to do this on initialization because this component works by calling all the component on_initialize_simulants (the chicken and egg Patrick referred to at mid day yesterday). This also allows us to easily go back to having simulants be born on different time steps. 
@mattkappel That's what I was hoping for but the code actually errors out instead :/